pkg/userdata/bottlerocket/node_userdata.go

```diff
@@ -75,15 +109,31 @@ func generateBottlerocketNodeUserData(bootstrapContainerUserData []byte, users [
 	bottlerocketInput := &bottlerocketSettingsInput{
 		BootstrapContainerUserData: b64BootstrapContainerUserData,
 		AdminContainerUserData:     b64AdminContainerUserData,
-		BootstrapContainerSource:   config.BootstrapImage,
-		PauseContainerSource:       config.PauseImage,
+		BootstrapContainerSource:   config.BottlerocketConfig.BootstrapImage,
+		PauseContainerSource:       config.BottlerocketConfig.PauseImage,
+	}
+
+	if config.Proxy != nil {
+		bottlerocketInput.HTTPSProxyEndpoint = config.Proxy.HTTPSProxy
+		if len(config.Proxy.NoProxy) > 0 {
```

Comment:
nit: do you need this? even if the array is empty or nil, the for loop should already take care of that

Reply:
yup don't need this, I think it got left after some trial-error to get this working